### PR TITLE
Fix #8285: Add you-tube warning popover

### LINKED ISSUE #8285: Add disable YouTube shields popover
### Description: 
According to the following spec: 
https://docs.google.com/document/d/1wSAowffI_-KnBci8CHtT-f7hw-_YgTSj-nVkiU4tIZw/edit

In a nutshell, we want to add a popover when the user visits YouTube and receives a "detected ad blocker warning" In this popover, we are informing them we suggest they disable adblocking with a Go ahead/no thanks buttons and additional details why and informing them about our brave-player

Essentially we are replacing youtube's warning with our own warning.

Figma:
https://www.figma.com/file/4tLGBHZNP8dL0vvphXpsd3/Brave-Player-for-YouTube?type=design&node-id=35-11529&mode=design&t=0KpevXC93Cc07ABV-0

### Test
  1. Enable ad-blockers for YouTube
  2. Go to YouTube.com on an iPad
  3. ~~Make sure your'e logged in. browse and watch videos until you see an Adblock warning from youtube. (note: i'm not sure if we will be blocking this warning so you might not see it but instead you will just jump straight to step 4)~~ This will be done at a later PR. For now skip to step 4
  4. See brave's popover. 
  5. This popover should appear once only and respect the users decision (i.e. shields are disabled on YouTube if the user selects the "disable" option. Note, the user may re-enable shields if they wish at a later time

The problem with this test is that for step 3,  nobody really knows how long it will take to get the warning. I myself never got the warning. So the warning seems to only affect some users, perhaps its AB testing or maybe longterm ad-block users get it. Nobody really knows. There are no concrete ways to reproduce step 3.

### PR DESCRIPTION
**Pull request by cuba**
## Summary of Changes

This pull request fixes #8285

## Submitter Checklist:

- [ ] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:
See issue for STR

## Screenshots:
Uploading Simulator Screen Recording - iPhone 15 Pro - 2023-10-25 at 10.47.04.mp4…

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Paged/CosmeticFiltersScriptHandler.swift [hunk 1]
```diff
@@ -83,6 +87,19 @@ class CosmeticFiltersScriptHandler: TabContentScript {
           }
         }
         
+        // Based on what we block we want to show the youtube warning
+        // TODO: @JS Figure out which selectors should trigger this warning
+        if !ShieldPreferences.hasSeenAntiAdBlockWarning.value {
```
Comment (soner-yuksel) [inline Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Paged/CosmeticFiltersScriptHandler.swift:92]: So this alert will be shown every time user open youtube on a new tab page even user dismisses the pop-over. Only way not to see this is to press one of the buttons now.

I believe this requirement is wrong and should not be working this way, where we harass the user with this new pop-over.

File: Sources/Brave/Frontend/Browser/BrowserViewController.swift [hunk 2]
```diff
@@ -2624,15 +2624,34 @@ extension BrowserViewController: TabDelegate {
...
+        
+        viewController.dismissCallback = { [weak self, weak popover] needsReload in
+          if needsReload {
+            self?.tabManager.reloadSelectedTab()
+          }
+          
+          popover?.dismissPopover()
+        }
+        
+        popover.arrowDirectionBehavior = .automatic
```
Comment (soner-yuksel) [inline Sources/Brave/Frontend/Browser/BrowserViewController.swift:2640]: Thanks for adding the arrow indicator, it looks better and more logical with it.

File: Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Paged/CosmeticFiltersScriptHandler.swift (hunk 1 above)
Comment (cuba) [reply Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Paged/CosmeticFiltersScriptHandler.swift:92]: No, I just haven't figured out how to check if the anti-adblock warning was seen yet. But this should only appear once that condition is met. But I can't reproduce the anti-adblock warning so I'm a little bit stuck, I removed that condition from this PR for now and will add it as a separate PR. 
And once the user choses an option, yes or no, then it won't appear ever again.

**Comment by cuba** [comment]
All combined here: https://github.com/brave/brave-ios/pull/8416

**Pull request closed without merging**